friends/src/App.js

```diff
@@ -0,0 +1,87 @@
+import React, {Component} from 'react';
+import './App.css';
+import axios from 'axios';
+import Friend from './Friend.js';
+import Form from './Form.js';
+import {Route} from 'react-router-dom';
+
+class App extends Component {
+  constructor(){
+    super();
+    this.state = {
+      data: [],
+      id: 0,
+      name: '',
+      age: 0,
+      email: ''
+    }
+  }
+
+  componentDidMount = () => {
+    axios.get('http://localhost:5000/friends')
+      .then( response => {
+        if(typeof response.data.message === 'string'){
+          Promise.reject("Error: Friends not found :( ")
+        }
+        this.setState({ data: response.data})
+      })
+      .catch( err=> console.log(err))
+    }
+
+  handleChange = (event)=>{
+    this.setState({
+      [event.target.id]: event.target.value
+    })
+  }
+
+  preventDefault = (e) => {
+    e.preventDefault();
+  }
+
+  submit = (e) => {
+    e.preventDefault();
+    let newID = this.state.data.length+1;
+    console.log(newID);
+    this.setState({
+      id: newID
+    })
+
+    const newObj = {
+        id: this.state.id,
+        name: this.state.name,
+        age: this.state.age,
+        email: this.state.email
+    }
+
+    axios.post('http://localhost:5000/friends', newObj)
```

Comment:
If you want your data to automatically update in the UI when you update the state of the data in the post request, look into the ComponentDidUpdate React Lifecycle Hook